src/Share/Survey/SurveyHelper.cs

```diff
@@ -0,0 +1,360 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Newtonsoft.Json;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.PowerShell.Share.Survey
+{
+    using Condition = Func<SurveyHelper, string, int, bool>;
+    using UpdateModule = Action<SurveyHelper, string, int>;
+
+    public class SurveyHelper
+    {
+        private const int CountExpiredDays = 30;
+        private const int LockExpiredDays = 30;
+        private const int SurveyTriggerCount = 3;
+        private const int FlushFrequecy = 5;
+        private const int DelayForSecondPrompt = 2;
+        private const int DelayForThirdPrompt = 5;
+
+        private static SurveyHelper Instance;
+
+        private int FlushCount;
+
+        private static string SurveyScheduleInfoFile = Path.Combine(
+            Environment.GetFolderPath(Environment.SpecialFolder.UserProfile),
+            ".Azure", "AzureRmSurvey.json");
+
+        public const string AzurePSInterceptSurvey = "Azure_PS_Intercept_Survey";
+
+        public const string Predictor = "Az.Predictor";
+
+        private DateTime LastPromptDate { get; set; }
+
+        private ConcurrentDictionary<string, ModuleInfo> Modules { get; set; }
+
+        private bool IgnoreSchedule;
+
+        private bool IsDisabledFromEnv => "Disabled".Equals(Environment.GetEnvironmentVariable(AzurePSInterceptSurvey), StringComparison.OrdinalIgnoreCase)
+                                        || "False".Equals(Environment.GetEnvironmentVariable(AzurePSInterceptSurvey), StringComparison.OrdinalIgnoreCase);
+
+        private readonly string CurrentDate;
+
+        private readonly DateTime Today;
+
+        private SurveyHelper()
+        {
+            CurrentDate = DateTime.UtcNow.ToString("yyyy-MM-dd");
+            Today = Convert.ToDateTime(CurrentDate);
```

Comment:
`DateTime.Today` returns today.
`CurrentDate` and `Today` are the same data but in different format. Do we need to keep the two?

Reply:
these two variables will be used quite often at runtime